config/run-axe.js

```diff
@@ -0,0 +1,152 @@
+
+const REMOTE_CHROME_URL = process.env['REMOTE_CHROME_URL'];
+const AXE_JS = fs.readFileSync(__dirname + '/../node_modules/axe-core/axe.js');
+const PAGES = [
+  '/',
+  '/page-templates/landing/',
+  '/page-templates/docs/',
+];
+
+function launchChromeLocally(headless=true) {
+  return chromeLauncher.launch({
+    chromeFlags: [
+      '--window-size=412,732',
+      '--disable-gpu',
+      headless ? '--headless' : ''
+    ]
+  });
+}
+
+function getRemoteChrome() {
+  const info = urlParse(REMOTE_CHROME_URL);
+  if (info.protocol !== 'http:')
+    throw new Error(`Unsupported protocol: ${info.protocol}`);
+  return new Promise(resolve => {
+    resolve({
+      host: info.hostname,
+      port: info.port,
+      kill() { return Promise.resolve(); }
+    });
+  });
+}
+
+// This function is only here so it can be easily .toString()'d
+// and run in the context of a web page by Chrome. It will not
+// be run in the node context.
+function runAxe() {
+  return new Promise((resolve, reject) => {
+    window.axe.run((err, results) => {
+      if (err) return reject(err);
+      resolve(JSON.stringify(results.violations));
+    });
+  });
+}
+
+let getChrome = REMOTE_CHROME_URL ? getRemoteChrome : launchChromeLocally;
+
+Promise.all([runServer(), getChrome()]).then(([server, chrome]) => {
+  const chromeHost = chrome.host || 'localhost';
+  console.log(`Static file server is listening at ${server.url}.`);
+  console.log(`Chrome is debuggable on http://${chromeHost}:${chrome.port}.`);
+  console.log(`Running aXe on:`);
+
+  CDP({
+    host: chrome.host,
+    port: chrome.port,
+  }, client => {
+    const {Page, Network, Runtime} = client;
+    const pagesLeft = PAGES.slice();
+    const loadNextPage = () => {
+      if (pagesLeft.length === 0) {
+        console.log(`Finished visiting ${PAGES.length} pages with no errors.`);
+        terminate(0);
+      } else {
+        const page = pagesLeft.pop();
+        const url = `${server.url}${page}`;
+        process.stdout.write(`  ${page} `);
+        Page.navigate({url});
+      }
+    };
+    const terminate = exitCode => {
+      client.close().then(() => {
+        chrome.kill().then(() => {
+          // Note that we're not killing the server; this is because
+          // the remote chrome instance (if we're using one) may be
+          // keeping some network connections to the server alive, which
+          // makes it harder to kill, so it's easier to just terminate.
+          const color = exitCode === 0 ? chalk.green : chalk.red;
+          console.log(color(`Terminating with exit code ${exitCode}.`));
+          process.exit(exitCode);
+        });
+      });
+    };
+
+    process.on('unhandledRejection', (reason, p) => {
+      console.log('Unhandled Rejection at:', p, 'reason:', reason);
+      terminate(1);
+    });
+
+    Promise.all([
+      Page.enable(),
+      Network.enable(),
+    ]).then(() => {
+      Network.responseReceived(({response}) => {
```

Comment:
Hmm, so I thought about it a bit, and realized that maybe it's actually good that it's counting any sub-resource failure as a 404 too.  For example, if it's 404'ing on CSS or JS, then that changes the accessibility profile of the page--calculated color contrast ratios may be incorrect, or certain ARIA attributes may not be present on the evaluated page.

Another thing is that the crawler from #321 isn't actually 100% accurate--the library we're using uses regular expressions to "scrape" resources for URLs, so it's actually possible that it might miss sub-resources that are noticed by actually loading a page in Chrome. So it might still be good to hard fail on sub-resource failures for the sake of redundancy, too.

What do you think? At the very least, I think we should log warnings on sub-resource fails.